### PR TITLE
document plugin helpers methodology

### PR DESCRIPTION
**Pull request by lieut-data**
This addition to the plugin documentation explains the purpose of plugin helpers and guides developers on when to extend using same.

### CODE REVIEW

File: site/content/extend/plugins/helpers.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: Plugin Helpers
+date: 2019-09-30T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 100
+---
+
+The plugin architecture consists of an [API](API), [hooks](http://localhost:1313/extend/plugins/server/reference/#Hooks), and plugin helpers.
```
Comment (amyblais) [inline site/content/extend/plugins/helpers.md]: ``hooks`` link doesn't work.

File: site/content/extend/plugins/helpers.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	GetUsers(options *model.UserGetOptions) ([]*model.User, *model.AppError)
+```
+
+Old servers won't do anything with new, unrecognized fields, but also won't break if they are present. When adding new options that will only be supported by some Mattermost server versions, consider wrapping the functionality with a plugin helper to help plugin authors safely use the API.
+
+## How do I add a plugin helper?
+
+To add a helper to the official [github.com/mattermost/mattermost-server/plugin](https://godoc.org/github.com/mattermost/mattermost-server/plugin#Helpers) package:
+
+* Extend the [`Helpers` interface](https://github.com/mattermost/mattermost-server/blob/master/plugin/helpers.go) with the new method
```
Comment (amyblais) [inline site/content/extend/plugins/helpers.md]:
```suggestion
* Extend the [`Helpers` interface](https://github.com/mattermost/mattermost-server/blob/master/plugin/helpers.go) with the new method.
```

File: site/content/extend/plugins/helpers.md (hunk 1 above)
Comment (lieut-data) [reply site/content/extend/plugins/helpers.md]: Oops! Thanks -- fixed :)

**Comment by hanzei** [comment]
I'm wondering is this file should be in https://developers.mattermost.com/extend/plugins/server/ instead, because it's just server plugins thing.

**Pull request merged**